src/axom/core/examples/core_containers.cpp

```diff
@@ -176,8 +170,103 @@ void demoArrayBasic()
   // _iteration_end
 }
 
+#ifdef __CUDACC__
+
+// _cuda_kernel_start
+// Aliases used for convenience
+using UnifiedIntArrayView = axom::ArrayView<int, 1, axom::MemorySpace::Unified>;
+using DeviceIntArrayView = axom::ArrayView<int, 1, axom::MemorySpace::Device>;
+
+__global__ void add(const UnifiedIntArrayView A,
+                    const UnifiedIntArrayView B,
+                    DeviceIntArrayView C)
+{
+  for(int i = 0; i < A.size(); i++)
+  {
+    C[i] = A[i] + B[i];
+  }
+}
+// _cuda_kernel_end
+
+#endif
+
+void demoArrayDevice()
+{
+  //This example requires Umpire
+#if defined(AXOM_USE_UMPIRE) && defined(AXOM_USE_CUDA) && defined(__CUDACC__)
+  // _cuda_array_create_start
+  constexpr int N = 10;
+  const int allocator_id = axom::getUmpireResourceAllocatorID(
+    umpire::resource::MemoryResourceType::Unified);
+
+  // By default the memory space can be specified at runtime.
+  // If this argument is not provided host memory will be allocated.
```

Comment:
This comment is a little confusing, because you're dealing both with a template parameter which specifies the memory space at compile time (with default value `Dynamic`) and also with a constructor parameter which specifies the memory allocator at runtime (with default value "the allocator for `Dynamic`", which is host memory).  Here's a stab at a rewrite, which may or may not match your intention:

```suggestion
  // The last template parameter specifies a memory space.
  // Its default value is Dynamic, which lets the user specify the
  // memory space at runtime with a memory allocator ID.  The
  // third constructor parameter specifies the allocator.
  // If this argument is not provided host memory will be allocated.
```